src/Compilers/Core/Portable/Symbols/SymbolEqualityComparer.cs

```diff
@@ -26,6 +26,9 @@ public sealed class SymbolEqualityComparer : IEqualityComparer<ISymbol?>
         // Internal only comparisons:
         internal readonly static SymbolEqualityComparer ConsiderEverything = new SymbolEqualityComparer(TypeCompareKind.ConsiderEverything);
 
+        // Ignore everything; for types, this reflects the identity conversion.
+        internal readonly static SymbolEqualityComparer IgnoreEverything = new SymbolEqualityComparer(TypeCompareKind.AllIgnoreOptions);
```

Comment:
>internal readonly static SymbolEqualityComparer IgnoreEverything = new SymbolEqualityComparer(TypeCompareKind.AllIgnoreOptions); [](start = 8, length = 128)

Compilers should not be using comparers that operate on ISymbol interfaces. CSharp compiler has its own symbol comparer that already provides this flavor for internal symbols.

Reply:
After looking through remaining parts of this PR, I think that this class shouldn't be modified.

---
In reply to: [362893135](https://github.com/dotnet/roslyn/pull/40677#discussion_r362893135) [](ancestors = 362893135)